opal_client/data/updater.py

```diff
@@ -233,5 +203,83 @@ async def wait_until_done(self):
         if self._subscriber_task is not None:
             await self._subscriber_task
 
+    @staticmethod
+    def calc_hash(data):
+        if not isinstance(data, str):
+            data = json.dumps(data)
+        return hashlib.sha256(data.encode('utf-8')).hexdigest()
+
+    async def report_update_results(self, update: DataUpdate, reports: List[DataEntryReport], data_fetcher: DataFetcher):
+        try:
+            whole_report = DataUpdateReport(update_id=update.id, reports=reports)
 
+            callbacks = update.callback.callbacks or opal_client_config.DEFAULT_UPDATE_CALLBACKS.callbacks
+            urls = []
+            for callback in callbacks:
+                if isinstance(callback, str):
+                    url = callback
+                    callback_config = opal_client_config.DEFAULT_UPDATE_CALLBACK_CONFIG.copy()
+                else:
+                    url, callback_config = callback
+                callback_config.data = whole_report.json()
+                urls.append((url, callback_config))
+
+            logger.info("Reporting the update to requested callbacks", urls=urls)
+            await data_fetcher.handle_urls(urls)
+            # TODO: log failed reports
```

Comment:
nit: this TODO does not seem so hard, simply add a log line?

Reply:
Its a little more than that- but sure I can do it